### PR TITLE
784 unit test sign_normalize

### PR DESCRIPTION
**Pull request by ttiaNlluu**
@DanilYachmenev Add `class Test_sign_normalize` to test `sign_normalize()` in `test_misc_transformations.py`.

1. Only test 1-dimensional DataFrame, do I need to test multi-dimensional DataFrame?
2. Should I test value '0' when `atol` = 0?
3. What other values do you want me to test?

### CODE REVIEW

File: core/signal_processing/test/test_misc_transformations.py [hunk 1]
```diff
@@ -10,6 +10,97 @@
 
 _LOG = logging.getLogger(__name__)
 
+class Test_sign_normalize(hunitest.TestCase):
+    """
+    Test that values in Series and DataFrame are normalized correctly
```
Comment (DanilYachmenev) [inline core/signal_processing/test/test_misc_transformations.py:15]: -> `Check that signal values are normalized according to sign correctly.`

File: core/signal_processing/test/test_misc_transformations.py [hunk 2]
```diff
@@ -10,6 +10,97 @@
...
+    """
+    def test_Series1(self) -> None:
```
Comment (DanilYachmenev) [inline core/signal_processing/test/test_misc_transformations.py:17]: use `test1`, `test2` etc, no need to reflect input signal type here

File: core/signal_processing/test/test_misc_transformations.py [hunk 3]
```diff
@@ -10,6 +10,97 @@
...
+        """
+        - Test on Series: all elements are in positive values
```
Comment (DanilYachmenev) [inline core/signal_processing/test/test_misc_transformations.py:19]:
```
- input signal is `pd.Series`
- atol = 0
```

good call to reflect input info in docstrings, maintain the format in all the further tests

File: core/signal_processing/test/test_misc_transformations.py [hunk 4]
```diff
@@ -10,6 +10,97 @@
...
+        - atol = 0
+        """
+        signal = pd.Series([1, 2, 3, 4, 5])
+        # atol = 0
```
Comment (DanilYachmenev) [inline core/signal_processing/test/test_misc_transformations.py:23]: no need, it is a default value

File: core/signal_processing/test/test_misc_transformations.py [hunk 5]
```diff
@@ -10,6 +10,97 @@
...
+        actual = csprmitr.sign_normalize(signal).all()
+        expected = pd.Series([1, 1, 1, 1, 1]).all()
```
Comment (DanilYachmenev) [inline core/signal_processing/test/test_misc_transformations.py:25]: why do you use `.all()`?
compare directly

File: core/signal_processing/test/test_misc_transformations.py [hunk 6]
```diff
@@ -10,6 +10,97 @@
...
+class Test_sign_normalize(hunitest.TestCase):
+    """
+    Test that values in Series and DataFrame are normalized correctly
+    """
+    def test_Series1(self) -> None:
+        """
+        - Test on Series: all elements are in positive values
+        - atol = 0
+        """
+        signal = pd.Series([1, 2, 3, 4, 5])
```
Comment (DanilYachmenev) [inline core/signal_processing/test/test_misc_transformations.py:22]: no need to test different signals with only positive, only negative etc

use `signal = pd.Series([-5, -1, 0, 1, 5])`
this will test it sufficiently

**Comment by DanilYachmenev** [comment]
@ttiaNlluu read `Save Reviewer time` section in `amp/docs/coding/all.submit_code_for_review.how_to_guide.md`, follow it

it is not less if not more important than your code itself rn
at first, your goal is to save your team leads' time

File: core/signal_processing/test/test_misc_transformations.py (hunk 5 above)
Comment (ttiaNlluu) [reply core/signal_processing/test/test_misc_transformations.py:25]: If I compare directly, it raise ValueError: 'The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.al...', let me know what's a better way to do.

Comment (DanilYachmenev) [reply core/signal_processing/test/test_misc_transformations.py:25]: here output is series/df - in this cases we use string signature approach
try
```
res = csprmitr.sign_normalize(signal)
actual_signature = hpandas.df_to_str(res)
expected_signature = r"""
...
"""
self.assert_equal(actual_signature, expected_signature, fuzzy_match=True)
```

you may not know `expected_signature` at first so put a random value there and run the test
it will give you the needed signature in error logs
update and run again

for reference and code style see `helpers/test/test_hpandas.py`

File: core/signal_processing/test/test_misc_transformations.py [hunk 7]
```diff
@@ -10,6 +10,97 @@
...
+    def test_DataFrame4(self) -> None:
+        """
+        - Test on DataFrame
+        - atol = 2
+        """
+        signal = pd.DataFrame([-4, -3, -2, -1, 0, 1, 2, 3, 4])
+        atol = 2
+        actual = csprmitr.sign_normalize(signal,atol)
+        expected = pd.DataFrame([-1, -1, -1, 0, 0, 0, 1, 1, 1])
+        self.assertTrue(actual.equals(expected))
```
[the remark this answers is not in the document]
Comment (ttiaNlluu) [reply core/signal_processing/test/test_misc_transformations.py:104]: Not sure if I run linter correctly. I run:
`invoke lint --files "./core/signal_processing/test/test_misc_transformations.py"`
And check that `linter_warnings.txt` is empty.
There are several warnings related to environment variables not being set, and saved in `linter_output.txt:`
`time="2024-04-17T11:07:04-04:00" level=warning msg="The \"AM_AWS_ACCESS_KEY_ID\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"AM_AWS_DEFAULT_REGION\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"AM_AWS_SECRET_ACCESS_KEY\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"AM_FORCE_TEST_FAIL\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"AM_TELEGRAM_TOKEN\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"CK_AWS_ACCESS_KEY_ID\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"CK_AWS_DEFAULT_REGION\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"CK_AWS_SECRET_ACCESS_KEY\" variable is not set. Defaulting to a blank string."
time="2024-04-17T11:07:04-04:00" level=warning msg="The \"CK_TELEGRAM_TOKEN\" variable is not set. Defaulting to a blank string."
Cannot connect to the Docker daemon at unix:///Users/tianlulu/.docker/run/docker.sock. Is the docker daemon running?`
Let me know if something I need to do.

[the remark this answers is not in the document]
Comment (samarth9008) [reply core/signal_processing/test/test_misc_transformations.py:104]: Docker process is not turned on in your local computer

[the remark this answers is not in the document]
Comment (ttiaNlluu) [reply core/signal_processing/test/test_misc_transformations.py:104]: Docker is running this time, 
and I run `invoke lint --files "./core/signal_processing/test/test_misc_transformations.py"` again. 
How to check the output to know it's successful?

**Comment by DanilYachmenev** [comment]
> Can I close this PR and open a new one? I created a branch under my forked repo, then linter failed with AssertionError. So, I create a branch from sorrentum/sorrentum, and I want to re-open a PR on that.

sure, do it if it helps
add a pointer to this PR and do not forget to delete the old branch
